CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -994,21 +994,24 @@ public static function updateOnNewPayment($recurringContributionID, $paymentStat
    * @param \CRM_Contribute_DAO_Contribution $contribution
    *
    * @throws \CRM_Core_Exception
-   * @throws \Civi\API\Exception\UnauthorizedException
    */
-  public static function updateOnTemplateUpdated(CRM_Contribute_DAO_Contribution $contribution) {
-    if (empty($contribution->contribution_recur_id)) {
+  public static function updateOnTemplateUpdated(CRM_Contribute_DAO_Contribution $contribution): void {
+    if ($contribution->is_template === '0' || empty($contribution->contribution_recur_id)) {
```

Comment:
Are you sure this will always come in as a string '0'?

Reply:
Yeah - the DAO layer does the string thing